.github/workflows/main.yml

```diff
@@ -1,64 +1,216 @@
-name: Scriptoria v1 CI
+name: Scriptoria v2 Test and Deploy
 
-# This is outdated. Reference for AWS deployment
 on:
-  # push:
-  # pull_request:
+  push:
+    branches:
+      - feature/svelte
+      - develop
+      - master
+  pull_request:
+    branches:
+      - feature/svelte
+      - develop
+      - master
   workflow_dispatch:
-    inputs:
-      logLevel:
-        description: "Log Level"
-        required: true
-        default: "warning"
-      tags:
-        description: "Tags"
 
 jobs:
-  build:
+  # Install dependencies once and cache for other jobs
+  setup:
     runs-on: ubuntu-latest
-    env:
-      AWS_ACCESS_KEY_ID: ${{ secrets.DEPLOY__AWS_ACCESS_KEY_ID }}
-      AWS_SECRET_ACCESS_KEY: ${{ secrets.DEPLOY__AWS_SECRET_ACCESS_KEY }}
-      AWS_DEFAULT_REGION: ${{ secrets.DEPLOY__AWS_DEFAULT_REGION }}
-      AWS_ECR_ACCOUNT: ${{ secrets.DEPLOY__AWS_ECR_ACCOUNT }}
-
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
-          node-version: 14
+          node-version: 24
+          cache: 'npm'
+          cache-dependency-path: package-lock.json
 
       - name: Get environment info
         run: |
-          echo Python version
-          python --version
-          pip --version
-          echo
           echo Node version
           node --version
-          yarn --version
           npm --version
           echo
           echo Docker version
           docker --version
           echo
-          echo Env
+          echo Environment variables
           env
 
-      - name: Frontend linting
-        run: ./run yarn && ./run yarn lint --quiet
+      - name: Restore node_modules cache
+        id: restore-cache
```

Comment:
Is this combine with the caching of npm above?

Reply:
This handles caching node_modules which is not cached by the setup-node cache step. That step handles `.npm/` metadata files instead.